### PR TITLE
fix(base-cluster/flux): adjust schema for flux gitRepositories

### PR DESCRIPTION
**Pull request by cwrau**
this tightens the validation so more edge cases are handled and are failing properly

### CODE REVIEW

File: charts/base-cluster/ci/flux-gitrepositories-values.yaml [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+      gitInterval: 5m
+    main57:
+      url: ssh://git@github.com/group/name.git
+      decryption:
+        provider: sops
+    main58:
+      url: ssh://git@github.com/group/name.git
+      gitInterval: 5m
+    main59:
+      url: ssh://git@github.com/group/name.git
```
[the remark this answers is not in the document]
Comment (marvinWolff) [reply charts/base-cluster/ci/flux-gitrepositories-values.yaml:319]: This is just a ci-test the ci, isn't it?

Why that many repositories? :man_shrugging:

[the remark this answers is not in the document]
Comment (cwrau) [reply charts/base-cluster/ci/flux-gitrepositories-values.yaml:319]: Yes, this is for testing.

To catch all the various edge cases we need to test them.

I've created https://github.com/teutonet/teutonet-helm-charts/issues/343 to better work with these scenarios

**Review by marvinWolff — APPROVED** [review]
